YamlCreate small tweaks

OfficialEsco opened the pull request:
+ WinGet validate will run after the manifest have been created, updated and new locale is created
+ Added Natural sorting so it actually picks the last version
+ Tags can now be Cross-Platform instead of having to be Cross Platform

- [X] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [X] Does your manifest conform to the [1.0 schema](https://github.com/microsoft/winget-cli/blob/master/doc/ManifestSpecv1.0.md)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----


###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/11753)

OfficialEsco committed:
Added WinGet Validate

OfficialEsco committed:
Added Natural sorting

OfficialEsco committed:
Fixed support for hyphen

OfficialEsco committed:
Edited max trim combines

OfficialEsco committed:
Ignore PackageIdentifier line

Issue spotted when updating the "Microsoft.Office" manifest

jedieaston commented:
This is unrelated to these changes specifically (I can open an issue if you want to deal with this in your next PR), but when I run the script under PowerShell 5.1, `Out-File -Path $LocaleManifestPath -Encoding 'UTF8'` fails with an error that -Path isn't a valid argument. 

Is there a good way to refactor the script to keep compatibility or shall we add a check to ensure the user is running it using PowerShell Core?

OfficialEsco commented:
> PowerShell 5.1, `Out-File -Path $LocaleManifestPath -Encoding 'UTF8'` fails with an error that -Path isn't a valid argument.

Oke... The inconsistencies between Powershell 5 and Powershell Core/7 is starting to piss me off..
I believe this can be solved by not having -Path?

Thank you for testing :))

OfficialEsco committed:
Out-File Fix

OfficialEsco commented:
Why does
`New-Variable -Name "FileExtensions" -Value ($FileExtensions.Split("- ").Trim()[1..257] -join ", ") -Scope Script -Force` break it
and not
`New-Variable -Name "FileExtensions" -Value ($FileExtensions.Split("-").Trim()[1..257] -join ", ") -Scope Script -Force` ????

OfficialEsco committed:
Fixed Split for Powershell 5

OfficialEsco commented:
Hey @jedieaston please quality check it now :)
APPARENTLY Powershell 5 `-split` and `.split` works differently.

OfficialEsco committed:
Fixed tags split for NewLocale

palenshus commented on Tools/YamlCreate.ps1:
```diff
@@ -141,7 +148,8 @@ Function Read-PreviousWinGet-Manifest {
                     $Tags = $FetchTags.Substring(0, $FetchTags.LastIndexOf(' '))
-                    New-Variable -Name "Tags" -Value ($Tags.Split("-").Trim()[1..100] -join ", ") -Scope Script -Force
+                    $Tags = $Tags -Split '- '
+                    New-Variable -Name "Tags" -Value ($Tags.Trim()[1..100] -join ", ") -Scope Script -Force
```
only 16 tags allowed

OfficialEsco replied:
Alright alright, fair play

OfficialEsco committed:
Fixed NewLocale max Tags split

OfficialEsco committed:
Added more InstallerTypes

SteveL-MSFT commented:
@OfficialEsco regarding `Out-File` differences, `-FilePath` is the original parameter in WinPS 5.1, but for PowerShell 7, we took a community PR to give it a `-Path` alias to the same parameter to make it consistent with other cmdlets that have a `-Path`.  We don't backport changes to WinPS5.1 except for security fixes.

As for `-split` vs `.split()` the latter is the .NET method which takes a string whereas `-split` the operator takes a regex.
